docker-compose.yml

```diff
@@ -140,6 +140,92 @@ services:
       - POSTGRES_HOST=postgres-veritable-cloudagent-bob
       - WALLET_ID=bob
       - WALLET_KEY=bob-key
+      
+  # -------------------- charlie -------------------------------#
+  veritable-ui-charlie:
+    container_name: veritable-ui-charlie
+    build:
+      context: .
+      dockerfile: Dockerfile
+    restart: always
+    depends_on:
+      - postgres-veritable-ui-charlie
+      - veritable-cloudagent-charlie
+    ports:
+      - 3002:3000
+    command: >
+      sh -c "
+        npm i -g pino-colada
+        node ./node_modules/.bin/knex migrate:latest
+        npm start | pino-colada"
+    env_file:
+      - docker/test.env
+      - .env
+    environment:
+      - NODE_ENV=production
+      - LOG_LEVEL=trace
+      - DB_HOST=postgres-veritable-ui-charlie
+      - DB_NAME=veritable-ui
+      - API_SWAGGER_TITLE=charlie
+      - API_SWAGGER_BG_COLOR=#ff3131
+      - PUBLIC_URL=http://localhost:3002
+      - CLOUDAGENT_ADMIN_ORIGIN=http://veritable-cloudagent-charlie:3000
+      - CLOUDAGENT_ADMIN_WS_ORIGIN=ws://veritable-cloudagent-charlie:3000
+      - COMPANY_HOUSE_API_URL=https://api.company-information.service.gov.uk
+      - COOKIE_SESSION_KEYS=secret
+      - DB_PASSWORD=postgres
+      - DB_USERNAME=postgres
+      - IDP_CLIENT_ID=veritable-ui
+      - IDP_PUBLIC_URL_PREFIX=http://localhost:3080/realms/veritable/protocol/openid-connect
+      - IDP_INTERNAL_URL_PREFIX=http://keycloak:8080/realms/veritable/protocol/openid-connect
+      - INVITATION_FROM_COMPANY_NUMBER= -> 07964699
+      - INVITATION_PIN_SECRET=secret
+      - ISSUANCE_DID_POLICY=EXISTING_OR_NEW
+      - ISSUANCE_SCHEMA_POLICY=EXISTING_OR_NEW
+      - ISSUANCE_CRED_DEF_POLICY=EXISTING_OR_NEW
+      - DEMO_MODE=true
+      - SMTP_HOST=smtp4dev
+      - SMTP_PASS=
+      - SMTP_PORT=25
+      - SMTP_USER=
+  postgres-veritable-ui-charlie:
+    image: postgres:17.0-alpine
+    container_name: postgres-veritable-ui-charlie
+    ports:
+      - 5434:5432
+    volumes:
+      - postgres-veritable-ui-charlie:/var/lib/postgresql/data
+    environment:
+      - POSTGRES_PASSWORD=postgres
+      - POSTGRES_USER=postgres
+      - POSTGRES_DB=veritable-ui
+  postgres-veritable-cloudagent-charlie:
+    container_name: postgres-veritable-cloudagent-charlie
+    image: postgres:17.0-alpine
+    restart: on-failure
+    volumes:
+      - postgres-veritable-cloudagent-charlie:/var/lib/postgresql/data
+    environment:
+      - POSTGRES_USER=postgres
+      - POSTGRES_PASSWORD=postgres
+      - POSTGRES_DB=postgres-veritable-cloudagent
+  veritable-cloudagent-charlie:
+    image: digicatapult/veritable-cloudagent
+    container_name: veritable-cloudagent-charlie
+    restart: always
+    depends_on:
+      ipfs:
+        condition: service_healthy
+    ports:
+      - 3102:3000
+    env_file:
+      - docker/cloudagent.env
+    environment:
+      - ENDPOINT=ws://veritable-cloudagent-charlie:5003
```

Comment:
Do all the agents' websocket ports point to 5003?

Reply:
yes! also a new observation.

Reply:
smtp4dev is on 5001 and 5000 is used by Apple AirPlay if you're going to change them

Reply:
yep all 5003 intentionally